docs/fixing-warnings.md

```diff
@@ -0,0 +1,186 @@
+
+# Trim warnings in .NET 6
+
+In .NET Core 3.1 and 5.0 we introduced trimming as a new preview feature for self-contained .NET
+core applications. Conceptually, the feature is very simple: when publishing the application, the
+.NET SDK analyzes the entire application, including the .NET framework, and removes all unused
+code. In the time trimming has been in preview, we've learned that trimming is very powerful --
+it can reduce application size by half, or more. However, we've also learned about the
+difficulties in safely trimming applications.
+
+The most difficult question in trimming is what is unused, or more precisely, what is used. For
+most standard C# code this is trivial -- the trimmer can easily walk method calls, field and
+property references, etc, and determine what code is used. Unfortunately, some features, like
+reflection, present a significant problem. Consider the following code:
+
+```C#
+string s = Console.ReadLine();
+Type type = Type.GetType(s);
+foreach (var m in type.GetMethods())
+{
+    Console.WriteLine(m.Name);
+}
+```
+
+In this example, `Type.GetType` dynamically requests a type with an unknown name, and then prints
+the names of all of its members. Because there's no way to know at publish time what type name is
```

Comment:
Nit: Since there is a GetMembers() that gets the names of all of its members, we may want to rephrase it as "all of its methods" or "all of its method members" instead